src/services/dynamo-database-service.ts

```diff
@@ -40,61 +40,28 @@ export class DynamoDatabaseService {
    * @param userId - the userId that comes from the request
    */
   public async retrieveRecordsByUserId(userId: string) {
-    logger.debug(`${LOGS_PREFIX_SENSITIVE_INFO} Attempting request to dynamo db, with ID : ${userId}`);
-    const parameters: QueryCommandInput = {
-      TableName: this.tableName,
-      KeyConditionExpression: '#pk = :id_value',
-      ExpressionAttributeNames: { '#pk': 'pk' },
-      ExpressionAttributeValues: { ':id_value': { S: userId } },
-      ProjectionExpression: 'blocked, suspended, resetPassword, reproveIdentity, sentAt, appliedAt, isAccountDeleted',
-    };
-
+    const parameters = this.inputParametersForDatabaseQuery(userId);
+    parameters.ProjectionExpression =
+      'blocked, suspended, resetPassword, reproveIdentity, sentAt, appliedAt, isAccountDeleted';
     const response: QueryCommandOutput = await this.dynamoClient.send(new QueryCommand(parameters));
-    if (!response.Items) {
-      const errorMessage = 'DynamoDB may have failed to query, returned a null response.';
-      logger.error(errorMessage);
-      logAndPublishMetric(MetricNames.DB_QUERY_ERROR_NO_RESPONSE);
-      throw new Error(errorMessage);
-    }
-
-    if (response.Items.length > 1) {
-      const errorMessage = 'DynamoDB returned more than one element.';
-      logger.error(errorMessage);
-      logAndPublishMetric(MetricNames.DB_QUERY_ERROR_TOO_MANY_ITEMS);
-      throw new TooManyRecordsError(errorMessage);
+    this.validateQueryResponse(response);
+    if (response.Items) {
+      return response.Items[0] ? (unmarshall(response.Items[0]) as DynamoDBStateResult) : undefined;
     }
-    return response.Items[0] ? (unmarshall(response.Items[0]) as DynamoDBStateResult) : undefined;
   }
 
   /**
    * Function to retrieve the full record from DynamoDB
    * @param userId - user ID passed in via path params
-   * @returns - record from dynamoDB
+   * @returns - unmarshalled record from dynamoDB or undefined
    */
   public async queryRecordFromDynamoDatabase(userId: string) {
```

Comment:
could it be helpful to rename this to something like `getFullAccountInformation` ?